gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java

```diff
@@ -181,13 +221,52 @@ protected HiveDataset computeNext() {
     };
   }
 
+
+  /**
+   * Deprecated. Use {@link #createHiveDataset(Table, Config)} instead
+   */
+  @Deprecated
   protected HiveDataset createHiveDataset(Table table) throws IOException {
-    return new HiveDataset(this.fs, this.clientPool, new org.apache.hadoop.hive.ql.metadata.Table(table),
-        this.properties);
+    return createHiveDataset(table, ConfigFactory.empty());
+  }
+
+  protected HiveDataset createHiveDataset(Table table, Config datasetConfig) throws IOException {
+    return new HiveDataset(this.fs, this.clientPool, new org.apache.hadoop.hive.ql.metadata.Table(table), this.properties, datasetConfig);
   }
 
   @Override
   public Path commonDatasetRoot() {
     return new Path("/");
   }
+
+  /**
+   * Gets the {@link Config} for this <code>dbAndTable</code>.
+   * Cases:
+   * <ul>
+   * <li>If {@link #configStoreUri} is available it gets the dataset config from the config store at this uri
+   * <li>If {@link #configStoreUri} is not available it uses the job config as dataset config
+   * <li>If {@link #datasetConfigPrefix} is specified, only configs with this prefix is returned
+   * <li>If {@link #datasetConfigPrefix} is not specified, all configs are returned
+   * </ul>
+   * @param dbAndTable of the dataset to get config
+   * @return the {@link Config} for <code>dbAndTable</code>
+   */
+  private Config getDatasetConfig(DbAndTable dbAndTable) throws ConfigStoreFactoryDoesNotExistsException,
+      ConfigStoreCreationException, URISyntaxException {
+
+    Config datasetConfig;
+
+    // Config store enabled
+    if (this.configStoreUri.isPresent()) {
+      datasetConfig = this.configClient.getConfig(this.configStoreUri.get() + HiveConfigClientUtils.getDatasetUri(dbAndTable));
+
+    // If config store is not enabled use job config
+    } else {
+      datasetConfig = this.jobConfig;
+    }
+
+    return StringUtils.isBlank(this.datasetConfigPrefix) ? datasetConfig : ConfigUtils.getConfig(datasetConfig,
```

Comment:
I don't think so. However, if one wants to retain all ```datasetConfig``` instead of scoped ones, the ```datasetConfigPrefix``` can be empty.